### ISSUE TITLE
How should we handle milestones?

### ISSUE THREAD

**Issue by zoechi** [bwu-dart-graveyard/polymer_elements#20]
I would suggest simply ascending numbers maybe synchronized with the package verison number.
Any other ideas?
Maybe we could use names like "current", "next", "later" and rename them to version numbers after the milestone is reached and a new "next" is created (I expect renaming milestones is supported).

I found some ideas (but I absolutily want to avoid overengineering)
- http://programmers.stackexchange.com/questions/129714/how-to-manage-github-issues-for-priority-etc
- https://github.com/blog/831-issues-2-0-the-next-generation
- http://www.lullabot.com/blog/article/managing-projects-github (nice github introduction)
## Conclusion

after some research I would go for milestones like described above (current, next, later)
and labels like in the first SO answer (fist link) with some modifications:
### 'issue type' group
- type:bug
- type:feature
- type:idea
- type:invalid
- type:support
- type:task
### 'issue priority' group
- ~~prio:low~~
- ~~prio:normal~~
- ~~prio:high~~

I prefer
- prio:blocking (drop everything you’re working on, this needs to be fixed today (we also had “überblocking” for extremely severe issues)
- prio:critical (this should be the next thing to work on, should be fixed within a few days)
- prio:high (important to work on)
- ~~prio:normal~~ (implicit - no prio tag is normal)
- prio:low (if you have some spare time this would be nice to have.)
### 'issue status' group

(These labels describe an issue's state in a defined workflow.)
- status:confirmed
- status:deferred
- status:fix-committed (what would that be for?)
- status:in-progress
- status:incomplete (what would that be for, waiting for response?)
- status:rejected
- ~~status:resolved~~ (I would prefer to just close a resolved issue, or why would I want to set the resolved tag?)
### 'issue information' group
- info:feedback-needed
- info:help-needed
- info:progress-25
- info:progress-50
- info:progress-75
### 'version tag' group

(I dont't think we need those)
- ~~ver:1.x~~
- ~~ver:1.1~~

What do you think about that?

**Comment by ErikGrimes** [comment]
I like the milestones scheme.  It's simple and establishes a basic backlog.   We can set the content for current by reviewing the open issues each week.  Next represents our plan.  I think anything on fire will find it's way into next or current on it's own.  The default labels for github look reasonable.  We can always change them if we find there's a need.

**Comment by zoechi** [comment]
I miss the labels for
- task (instead of enhancement I would like to distinguish between feature request and task).
  This could be solved by replacing enhancement with task and feature or (feature-request)
- blocked (blocked by another bug or missing feature to easily see what should be revisited regularily)
- blocked-dartbug (to easily see what should be revisited when a new Dart version is available)
- feedback-needed (to be available to indicate a bug/task/... is blocked until feedback is provided) eventually blocked-needs-feedback would be a better name.

Adding more tags would make the tags confusing.
But using tag groups makes them more expressive and more clear.
Therefore my suggestion to introduce tag groups immediately instead of changing to much later.

I agree with your suggestion to omit further changes/additions until there is a need. 

What do you think?

**Comment by ErikGrimes** [comment]
I gave it a bit of thought and then went over to www.dartbug.com to see how
they do things.   I figure just adapt their scheme preemptively. Our issues
may live on in dartbug. That would give us the following labels.

type:codehealth
type:defect
type:documentation
type:enhancement
type:task
status:accepted
status:fixed
status:needsInfo
status:started
status:triaged
status:wontFix

We could add status:blocked and status:blocked-dartbug to that list.  Does
that work?  If so, I'll create the labels, apply them to the existing
issues, and close this issue.

**Comment by zoechi** [comment]
great idea to look at their repository
that would be perfect.

**Comment by ErikGrimes** [comment]
Added all the labels and tried to make the colors reasonable.
